src/coreclr/tools/Common/TypeSystem/IL/ILImporter.cs

```diff
@@ -447,8 +456,11 @@ private void ImportBasicBlock(BasicBlock basicBlock)
                     case ILOpcode.blt_un_s:
                         {
                             int delta = (sbyte)ReadILByte();
-                            ImportBranch(opCode + (ILOpcode.br - ILOpcode.br_s),
-                                _basicBlocks[_currentOffset + delta], (opCode != ILOpcode.br_s) ? _basicBlocks[_currentOffset] : null);
+                            if (_currentOffset < _basicBlocks.Length && _currentOffset + delta < _basicBlocks.Length)
+                            {
+                                ImportBranch(opCode + (ILOpcode.br - ILOpcode.br_s),
+                                    _basicBlocks[_currentOffset + delta], (opCode != ILOpcode.br_s) ? _basicBlocks[_currentOffset] : null);
+                            }
```

Comment:
Would the new if check ever be false given we'd abort for this in `FindJumpTargets` already?

Reply:
We don't abort at `ReportFallthroughAtEndOfMethod`. At least that was the one case of usage case, and I do not know the codebase that well to think about is this abort is reasonable way to handle.

Reply:
We may want to abort at `ReportFallthroughAtEndOfMethod` to keep things simple.